### PR TITLE
RR-23-Create-3-user-levels-as-User-Approver-Administrator

### PR DESCRIPTION
**Pull request by RajatMCAF**
[ADD] added user groups, record rule and category.

- Added (user, approver and admin)
- added category
- added basic record rule for admin and user (might need to change them in future)

- Currently implied_ids are commented.

### CODE REVIEW

File: resource_reservation/security/record_rule.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <field name="model_id" ref="model_resource_reservation"/>
+    <field name="domain_force">[('user_id', '=', user.id)]</field>
+    <field name="groups" eval="[(4, ref('group_resource_reservation_user'))]"/>
+    <field name="perm_read" eval="True"/>
+    <field name="perm_write" eval="False"/>
+    <field name="perm_unlink" eval="False"/>
+    <field name="perm_create" eval="False"/>
+</record>
+
+    <record id="record_rule_admin_resource_reservation" model="ir.rule">
```
Comment (MORPSmcaf) [inline resource_reservation/security/record_rule.xml:15]: align

File: resource_reservation/security/record_rule.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <field name="name">User Resource Reservation Rule</field>
+    <field name="model_id" ref="model_resource_reservation"/>
+    <field name="domain_force">[('user_id', '=', user.id)]</field>
+    <field name="groups" eval="[(4, ref('group_resource_reservation_user'))]"/>
+    <field name="perm_read" eval="True"/>
+    <field name="perm_write" eval="False"/>
+    <field name="perm_unlink" eval="False"/>
+    <field name="perm_create" eval="False"/>
+</record>
+
```
Comment (MORPSmcaf) [inline resource_reservation/security/record_rule.xml:14]: add approver

File: resource_reservation/__manifest__.py [hunk 3]
```diff
@@ -11,6 +11,8 @@
     'depends': ['base'],
     'data': [
         'security/ir.model.access.csv',
+        'security/security.xml',
```
Comment (MORPSmcaf) [inline resource_reservation/__manifest__.py:14]: rename like group

File: resource_reservation/__manifest__.py [hunk 4]
```diff
@@ -11,6 +11,8 @@
...
+        'security/record_rule.xml',
```
Comment (MORPSmcaf) [inline resource_reservation/__manifest__.py:15]: rename like a security + model name

File: resource_reservation/security/security.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    <record id="group_resource_reservation_approver" model="res.groups">
+        <field name="name">Approver</field>
+        <field name="category_id" ref="module_category_resource_reservation"/>
+<!--        <field name="implied_ids" eval="[(4, ref('group_resource_reservation_user'))]"/>-->
+    </record>
+
+        <record id="group_resource_reservation_admin" model="res.groups">
+        <field name="name">Administrator</field>
+        <field name="category_id" ref="module_category_resource_reservation"/>
+<!--        <field name="implied_ids" eval="[(4, ref('group_resource_reservation_approver'))]"/>-->
```
[the remark this answers is not in the document]
Comment (RajatMCAF) [reply resource_reservation/security/security.xml:25]: Complted

File: resource_reservation/security/record_rule.xml (hunk 1 above)
Comment (RajatMCAF) [reply resource_reservation/security/record_rule.xml:15]: Done

File: resource_reservation/__manifest__.py (hunk 3 above)
Comment (RajatMCAF) [reply resource_reservation/__manifest__.py:14]: done

File: resource_reservation/__manifest__.py (hunk 4 above)
Comment (RajatMCAF) [reply resource_reservation/__manifest__.py:15]: Done

File: resource_reservation/security/record_rule.xml (hunk 2 above)
Comment (RajatMCAF) [reply resource_reservation/security/record_rule.xml:14]: Done

**Pull request closed without merging**